redirect_cache.c

```diff
@@ -100,7 +100,7 @@ enum_func_status mysqlnd_azure_add_redirect_cache(const MYSQLND_CONN_DATA* conn,
 /* }}} */
 
 /* {{{ mysqlnd_azure_remove_redirect_cache */
-enum_func_status mysqlnd_azure_remove_redirect_cache(const MYSQLND_CONN_DATA* conn, const char* user, const char* host, int port)
+enum_func_status mysqlnd_azure_remove_redirect_cache(const char* user, const char* host, int port)
```

Comment:
One question. I see mysqlnd_azure_add_redirect_cache and mysqlnd_azure_remove_redirect_cache are called when connections are handled. Should mysqlnd_azure_add_redirect_cache and mysqlnd_azure_remove_redirect_cache be designed as thread-safe or not?

Reply:
The cache uses php's HashTable and is a module global of extension. PHP's module global is designed with consideration of handling thread-safe. It depends on compile configuration which is consistent with that PHP self.

Reply:
I can understand this, but how about  MYSQLND_AZURE_G(redirectCache) = mnd_pemalloc(sizeof(HashTable), 1); in add and zend_hash_str_del(MYSQLND_AZURE_G(redirectCache), key, strlen(key)); in remove? Will it lead to double-malloc and double-free when multiple connections enter in add and remove cache function?

Reply:
Same problem happened at mysqlnd_azure_get_is_using_redirect and mysqlnd_azure_set_is_using_redirect

Reply:
After double check, it will have no problem for multi-thread scenario, no change needed.

Reply:
I didn't read out the meaning from email that there was a decision. But after discussion and suggestion from Xiangyu and Christoph, Change tie cache to global will help for cache sharing. I've changed the design